apt/private/index.bzl

```diff
@@ -51,13 +52,16 @@ filegroup(
 '''
 
 def _deb_package_index_impl(rctx):
+    # Get the package name with either bzlmod enabled or disabled
+    package_name = rctx.attr.name.split("~")[-1]
+
     package_defs = [
         '"""Generated by rules_distroless. DO NOT EDIT."""',
         """load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")""",
         "",
         "",
         "# buildifier: disable=function-docstring",
-        "def {}_packages():".format(rctx.attr.name),
+        "def {}_packages():".format(package_name),
```

Comment:
I also had to get the last component of the path, as otherwise it would put ~ in the function name and then error: `syntax error at '~': expected (`